### PR TITLE
TST: use app, client fixture to reuse in tests

### PR DESCRIPTION
**Pull request by pratapvardhan**
@vincentsarago – thoughts on reusing fixtures? Reduces the need to keep defining client in every case.

### CODE REVIEW

File: src/titiler/core/tests/test_case_middleware.py [hunk 1]
```diff
@@ -2,71 +2,70 @@
...
+    return app
+
+
+@pytest.fixture
+def client(app):
+    """Create test client."""
+    return TestClient(app)
+
+
+def test_lowercase_middleware(app, client):
```
Comment (vincentsarago) [inline src/titiler/core/tests/test_case_middleware.py:27]: We don't need the `app` fixture here then

Comment (vincentsarago) [reply src/titiler/core/tests/test_case_middleware.py:27]: ah no we need it, But then it's a bit weird that we register the endpoint after creating the application 😬

Comment (pratapvardhan) [reply src/titiler/core/tests/test_case_middleware.py:27]: That's correct, we can leave it as it is and close this PR if the change doesn't seem clean enough.

Comment (vincentsarago) [reply src/titiler/core/tests/test_case_middleware.py:27]: Yeah I'm not convinced we need this change, sorry 🙏

**Pull request closed without merging**